Texture.py

```diff
@@ -1,163 +1,326 @@
-#FIXME: Move to xboxpy.NV2A.PGRAPH.Texture
+"""Provides functions for manipulating xbox textures."""
 
-from xboxpy import nv2a
+# pylint: disable=consider-using-f-string
+# pylint: disable=too-many-arguments
+# pylint: disable=too-many-locals
+# pylint: disable=too-many-branches
+# pylint: disable=too-many-statements
+
+# FIXME: Move to xboxpy.NV2A.PGRAPH.Texture
 
 from collections import namedtuple
 from PIL import Image
 
+from Xbox import Xbox
+import XboxHelper
+from xboxpy import nv2a
+
+# Value that may be added to contiguous memory addresses to access as ADDR_AGPMEM, which
```

Comment:
This is what I do in the pgraph test suite; relatively confident that it works correctly (since it does there, and so far has for all my test cases here), but if there are any known reasons not to take this approach please let me know.

Reply:
Interesting hack. I'm surprised if this works, because to my knowledge, it isn't a feature of AGP, but of the GPU memory subsystem (as far as I remember, each tile can be configured with the surface format directly).
But if it works, then this is probably a good solution.

Reply:
Did you check the depth-buffers?
You probably also have to disable compression on those, but I don't remember where I do that.

Ideally we'd be able to decompress on the PC side, but I don't know how this works (but I know there's at least *some* information about it in tech articles; maybe even nouveau).

Reply:
I have not checked them in nv2a-trace, but I definitely do use the same technique for [the pgraph test suite](https://github.com/abaire/nxdk_pgraph_tests/blob/7b93f25b248c7c01f6ebdd58b60b4f33c8e57813/src/test_host.cpp#L721) and it [appears to work reliably](https://github.com/abaire/nxdk_pgraph_tests_golden_results/wiki/Results-Depth_buffer) (for clarity "CompZ" and "Cy" in those results are with compression enabled).

Reply:
> This is what I do in the pgraph test suite; relatively confident that it works correctly (since it does there, and so far has for all my test cases here), but if there are any known reasons not to take this approach please let me know.

The main reason to avoid reading from this address range is performance